File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> hmap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {

            Integer index = hmap.get(nums[i]);

            if(index == null) {
                hmap.put(target - nums[i], i);
            } else if (index != null) {
                return new int[] {index, i};
            }
        }

        return new int[]{};
        
    }
}

File: 0001-two-sum/0001-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {

        //dictionary to store index and complement value
        var indexByValue = [Int:Int]()

        //iterate over the given array
        for(currentIndex, value) in nums.enumerated() {

            //hold the complement value
            let complement = target - value

            //if this complement alr seen then we know that the answer is ready to return
            if let complementIndex = indexByValue[complement] { //unwrapping
                
                return [complementIndex, currentIndex];

            }

            indexByValue[value] = currentIndex

        }

        return []


        
    }
}